classes/fields/pick.php

```diff
@@ -1305,6 +1305,9 @@ public function build_dfv_field_item_data_recurse_item( $item_id, $item_title, $
 		if ( ! empty( $args->value ) ) {
 			// The value may be a single non-array value.
 			$values = (array) $args->value;
+
+			// Cast values in array as string.
+			$values = array_map( 'strval', $values );
```

Comment:
This is still casting conditionally for no good reason that I can see.  It shouldn't functionally matter right now as the only other case is an empty array as the code stands, but code changes.  We insist on a strict string comparison on keys later in the code so we should always, unconditionally cast the keys IMO.

Reply:
The danger IME is someone comes along later and modifies the enclosing `if` so it's now `if (! empty( $args->value )  && now_only_executes_some_of_the_time )` and we start failing to cast again.

Reply:
Fixed